cmake/compilerFlagsExiv2.cmake

```diff
@@ -1,79 +1,121 @@
+
     if ( EXIV2_TEAM_WARNINGS_AS_ERRORS )
-        add_compile_options(-Werror -Wno-error=deprecated-declarations)
-    endif ()
+        if ( COMPILE_FLAG_WERROR )
+            add_compile_options(-Werror)
+        endif()
+        if ( COMPILE_ERROR_DEP )
+            add_compile_options(-Wno-error=deprecated-declarations)
+        endif()
+    endif()
 
+    # Note that this is intended to be used only by Exiv2 developers/contributors.
     if ( EXIV2_TEAM_EXTRA_WARNINGS )
-        # Note that this is intended to be used only by Exiv2 developers/contributors.
-
+        check_c_compiler_flag(-Wextra                    COMPILE_FLAG_WEXTRA            )
+        check_c_compiler_flag(-Wlogical-op               COMPILE_FLAG_WLOGICAL_OP       )
+        check_c_compiler_flag(-Wdouble-promotion         COMPILE_FLAG_WDOUBLE_PROMOTION )
+        check_c_compiler_flag(-Wshadow                   COMPILE_FLAG_WSHADOW           )
+        check_c_compiler_flag(-Wuseless-cast             COMPILE_FLAG_WUSELESS_CAST     )
+        check_c_compiler_flag(-Wpointer-arith            COMPILE_FLAG_WPOINTER_ARITH    )
+        check_c_compiler_flag(-Wformat=2                 COMPILE_FLAG_WFORMAT_2         )
+        check_c_compiler_flag(-Warray-bounds=2           COMPILE_FLAG_WARRAY_BOUNDS_2   )
+        check_c_compiler_flag(-Wduplicated-cond          COMPILE_FLAG_WDUPLICATED_COND  )
+        check_c_compiler_flag(-Wduplicated-branches      COMPILE_FLAG_WDUPLICATED_BRAN  )
+        check_c_compiler_flag(-Wrestrict                 COMPILE_FLAG_WRESTRICT         )
+        
         if ( COMPILER_IS_GCC )
-            if ( CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 4.0 )
-                string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}
-                    " -Wextra"
-                    " -Wlogical-op"
-                    " -Wdouble-promotion"
-                    " -Wshadow"
-                    " -Wuseless-cast"
-                    " -Wpointer-arith" # This warning is also enabled by -Wpedantic
-                    " -Wformat=2"
-                    #" -Wold-style-cast"
-                )
-            endif ()
-
-            if ( CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 5.0 )
-              string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}
-                " -Warray-bounds=2"
-                )
-            endif ()
-
-            if ( CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 6.0 )
-                string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}
-                    " -Wduplicated-cond"
-                )
-            endif ()
-
-            if ( CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.0 )
-                string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}
-                    " -Wduplicated-branches"
-                    " -Wrestrict"
-                )
+            if ( COMPILE_FLAG_WEXTRA )
+                string(CONCAT EXTRA_COMPILE_FLAGS ${EXTRA_COMPILE_FLAGS}" -Wextra")
```

Comment:
This results in strange quoting in the flags.make files, i.e.:

`CXX_FLAGS =   -fno-omit-frame-pointer -fsanitize=address,undefined -fno-sanitize-recover=all -fvisibility=hidden -fvisibility-inlines-hidden    " -Wextra"" -Wlogical-op"" -Wdouble-promotion"" -Wshadow"" -Wpointer-arith"" -Wformat=2"" -Warray-bounds=2"" -Wduplicated-cond"" -Wduplicated-branches"" -Wrestrict" -fstack-protector-strong -fstack-clash-protection -D_GLIBCXX_ASSERTIONS -Wall -Wcast-align -Wpointer-arith -Wformat-security -Wmissing-format-attribute -W`

I'm not sure whether gcc will use those oddly quoted flags or not.

It seems simpler to me to use `set` rather than `string(CONCAT`:

```
set(EXTRA_COMPILE_FLAGS "${EXTRA_COMPILE_FLAGS} -Wextra")
```